tests/test_matrix.py

```diff
@@ -342,3 +348,80 @@ def testFixSigns(self):
         x[:, 3] = 0
         y[:, 3] = 0
         self.assertTrue(np.allclose(fix_signs(x), y))
+
+
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_nearest_rotations(dtype):
+    n_rots = 5
+    rots = Rotation.generate_random_rotations(n_rots, seed=0, dtype=dtype).matrices
+
+    # Add some noise to the rotations.
+    noise = 1e-3 * randn(n_rots * 9, seed=0).astype(dtype, copy=False).reshape(
+        n_rots, 3, 3
+    )
+    noisy_rots = rots + noise
+
+    # Find nearest rotations for stack.
+    nearest_rots = nearest_rotations(noisy_rots)
+
+    # Check that estimates are rotation matrices.
+    _is_rotation(nearest_rots, dtype)
+
+    # Check that estimates are close to original rotations.
+    mean_aligned_angular_distance(rots, nearest_rots, degree_tol=1)
+
+    # Check dtype pass-through.
+    assert nearest_rots.dtype == dtype
+
+
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_nearest_rotations_reflection(dtype):
+    # Generate singleton rotation.
+    rot = Rotation.generate_random_rotations(1, seed=0, dtype=dtype).matrices[0]
+
+    # Add a reflection and some noise to the rotation.
+    refl = rot @ np.diag((1, -1, 1)).astype(dtype)
+    noise = 1e-3 * randn(9, seed=0).astype(dtype, copy=False).reshape(3, 3)
+    noisy_refl = refl + noise
+
+    # Find nearest rotation.
+    nearest_rot = nearest_rotations(noisy_refl)
+
+    # Check that estimate is a rotation.
+    _is_rotation(nearest_rot, dtype)
+
+    # Check that we retain singleton shape.
+    assert nearest_rot.shape == rot.shape
+
+
+def test_nearest_rotations_error():
+    # Check error for bad ndim.
+    A = randn(2 * 5 * 3 * 3, seed=0).reshape(2, 5, 3, 3)
```

Comment:
Think you can use `np.empty` here.  For this tiny array its not a problem though.  FWIW, generating really large random arrays can actually consume some compute time...  

![image](https://github.com/ComputationalCryoEM/ASPIRE-Python/assets/47759732/30b28498-d9d6-47fe-a917-7ce88e62421a)